src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -2772,6 +2773,47 @@ def load_arguments(self, _):
             action="store_true",
         )
 
+    with self.argument_context("aks agent") as c:
+        c.positional("prompt", help="Ask any question and answer using available tools.")
+        c.argument(
+            "max_steps",
+            type=int,
+            default=10,
+            required=False,
```

Comment:
You does set a default value of the corresponding parameter max_steps in function aks_agent, which is contrary to your declaration here. If you want to mark it as optional, please set a default value (like None or "") for the parameter.